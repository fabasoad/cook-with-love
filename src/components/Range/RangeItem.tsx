import * as React from 'react'

interface RangeItemProps {
  img: any
  name: string
  subtitle: string
  title: string
  url: string
}

export default function RangeItem(props: RangeItemProps) {
  return (
    <div className="col-sm-6 col-md-3 col-lg-3 web">
      <div className="range-item">
        <div className="hover-bg text-center">
          <a href={props.url} title={props.name} rel="noopener noreferrer" target="_blank">
            <div className="hover-text">
              <h4>{props.title}</h4>
              <small>{props.subtitle}</small>
              <i className="fa fa-search"></i>
            </div>
            <img className="img-thumbnail" src={props.img} alt={props.name} />
          </a>
        </div>
      </div>
    </div>
  )
}
