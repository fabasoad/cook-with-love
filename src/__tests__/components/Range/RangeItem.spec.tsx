/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import RangeItem from '../../../components/Range/RangeItem'

test('should render RangeItem correctly', () => {
  const wrapper: ShallowWrapper = shallow(<RangeItem
    img="test-img"
    name="test-name"
    title="test-title"
  />)
  expect(wrapper).toMatchSnapshot()
})
