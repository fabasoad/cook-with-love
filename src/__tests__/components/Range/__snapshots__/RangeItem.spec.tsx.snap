// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RangeItem correctly 1`] = `
<div
  className="col-sm-6 col-md-3 col-lg-3 web"
>
  <div
    className="range-item"
  >
    <div
      className="hover-bg text-center"
    >
      <div
        className="hover-text"
      >
        <h4>
          test-title
        </h4>
      </div>
      <img
        alt="test-name"
        className="img-thumbnail"
        src="test-img"
      />
    </div>
  </div>
</div>
`;
